### ISSUE TITLE
TOPICS: Add more than 50 emails at once?

### ISSUE THREAD

**Issue by BeccaMelhuish** [citizenos/citizenos-fe#1633]
**User insights:** 
In a Support query, @ssin1901 recently had an insight that an admin for an umbrella organisation was getting frustrated at only being able to add 50 emails at once.  (@ssin1901 Please add any more details if I've missed something). 

This backs up an insight from my user testing, in which an admin for a civic movement noted that 50 at a time wouldn't be enough for them. (She said adding more at a time would be more a 'nice to have', but if possible, would make things easier.). 

**Suggestions:** 
@ilmartyrk is it technically difficult to allow admins to upload more than 50 at once, ideally an unlimited number? Or what was the reason for limiting to 50 at a time?

@kevincrepin the UX of this could potentially be tackled in connection with [this issue](https://github.com/citizenos/citizenos-fe/issues/1531)? 

I wonder if we could also have an option for admins to upload an excel sheet of emails as a way to add all the emails very easily, if we specified whatever format might make this easier technically (i.e. all email adresses in one column, and nothing else in the sheet)?  

See also this connected issue: https://github.com/citizenos/citizenos-fe/issues/1647

**Comment by ssin1901** [comment]
@BeccaMelhuish yes, this was a major paint point as she had to add them in batches. It took a long time.

**Comment by BeccaMelhuish** [comment]
@ssin1901 I'll move this to 'Next?' if it was a major pain point, and we can discuss with Ilmar when back if there's a simple fix or not :)

**Comment by kevincrepin** [comment]
I think this bulk adding via xls or csv makes sense as an option. Maybe that wording can be improved @BeccaMelhuish . I think an option to download a template file (however simple it is) is useful in this case.

![Screenshot 2024-11-15 at 10 27 33](https://github.com/user-attachments/assets/41ba8659-33b1-4ef9-94d3-5c7da4a54557)

@ilmartyrk design is here: https://www.figma.com/design/gOJ4i00538e0qaP6IKIPVE/Platform-2.0---Design?node-id=524-62243&t=7QsMeQF7iK0Trubz-1

**Comment by ssin1901** [comment]
@BeccaMelhuish @ilmartyrk I was doing some research for the testing and the general guidelines were to avoid sending more than 50 emails at once. Here are the risks

1. If large volumes of emails bounce i.e user accidentally types the wrong email, our domain risks getting blacklisted
2. In general spam filters could get triggered by sending emails in bulk
3. Email providers like gmail provide throttling

I think rather than trying to fix the feature, we should nudge users towards using the share link.

**Comment by ilmartyrk** [comment]
@kevincrepin @BeccaMelhuish @ssin1901 Kevins design would work well, but I have to agree that making the sending of huge amount of e-mails at once will cause issues. We already have had issues with our newsletters so this could possibly happen again in this situation. Also this seems to be a "premium" feature and we are a free platform. So I totally support that we rather nudge users towards share links

**Comment by BeccaMelhuish** [comment]
@ilmartyrk @ssin1901 this all makes sense :) 

Though just to check @ilmartyrk, I guess there's no simple way to make sending the emails "all at once" for the user, but actually in practice our system only sends out 50 at a time? 

The downside to nudging them to use the share link is that they then can't very easily control who has access. So for something very official (like a general assembly) in which exactly the right people need to vote, this wouldn't work so well. 

However, such use cases aren't our target use cases, and we can't please everyone (especially as a free platform, as Ilmar mentioned). 

We _could_ add a line saying "Sending lots of invitations? Try using the share link." 

_But_ I also have the feeling that for anyone for whom that is a suitable solution (e.g. probably not the recent example that prompted this conversation) then they would easily think of that themselves, as users in tests have no trouble finding the link option. Maybe not, but that's my hunch. 

@kevincrepin what are your feelings on adding this prompt? 

But either way, I'm thinking we could come back and revisit this if it turns out to be important for our main use case focus, but not give it too much priority for now.

**Comment by BeccaMelhuish** [comment]
New user insight is that the organisation in question **would** have used the share link / QR code if they'd thought of it, and they noted this issue as a serious downside in their experience, costing them an 'enormous amount of time'. 

So in light of this, I think unless DUX have a better idea or an objection, then let's add this line indeed: 

"Sending lots of invitations? Try using the share link."

I have added it in the translation strings here:

![Image](https://github.com/user-attachments/assets/abe84683-1fd6-4e07-b86e-70fef33d75c5)

So will now move it to 'in progress' for Ilmar to merge :)

**Comment by ilmartyrk** [comment]
@BeccaMelhuish merged the translations

**Comment by BeccaMelhuish** [comment]
Checked in all relevant places in test and all good, will send to pre-live :)
